### PR TITLE
gh-107957: Fix `dir()` sample output in `enum` docs

### PR DESCRIPTION
**Pull request by ujan-r**
* Issue: gh-107957

----
:books: Documentation preview :books:: https://cpython-previews--107958.org.readthedocs.build/

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2023-08-14-20-38-39.gh-issue-107957.DU3mID.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix sample outputs for :func:`dir()` function in :mod:`enum` docs.
```
Comment (CharlieZhao95) [inline Misc/NEWS.d/next/Documentation/2023-08-14-20-38-39.gh-issue-107957.DU3mID.rst]: According to my experience, there is no need to add NEWS when we only modify the document, just wait for the triage member to add the `skip news` label.

File: Doc/howto/enum.rst [hunk 2]
```diff
@@ -1050,10 +1050,14 @@ If you give your enum subclass extra methods, like the `Planet`_
...
-    >>> dir(Planet.EARTH)                   # doctest: +SKIP
-    ['__class__', '__doc__', '__module__', 'mass', 'name', 'radius', 'surface_gravity', 'value']
+    >>> dir(Planet)
+    ['EARTH', 'JUPITER', 'MARS', 'MERCURY', 'NEPTUNE', 'SATURN', 'URANUS', 'VENUS', '__class__', '__contains__', '__doc__', '__getitem__', '__init_subclass__', '__iter__', '__len__', '__members__', '__module__', '__name__', '__qualname__']
+    >>> dir(Planet.EARTH)
+    ['EARTH', 'JUPITER', 'MARS', 'MERCURY', 'NEPTUNE', 'SATURN', 'URANUS', 'VENUS', '__class__', '__doc__', '__eq__', '__hash__', '__module__', 'mass', 'name', 'radius', 'surface_gravity', 'value']
+
+.. versionchanged:: 3.11
+    Additional ``__dunder__`` names are returned for enums and enum members.
+    Calling :func:`dir` on an enum member also returns the other members.
```
Comment (AA-Turner) [inline Doc/howto/enum.rst:1060]: We don't need versionchanged in a How To document 

```suggestion
```

File: Doc/howto/enum.rst [hunk 3]
```diff
@@ -1050,10 +1050,14 @@ If you give your enum subclass extra methods, like the `Planet`_
 class below, those methods will show up in a :func:`dir` of the member,
 but not of the class::
 
-    >>> dir(Planet)                         # doctest: +SKIP
-    ['EARTH', 'JUPITER', 'MARS', 'MERCURY', 'NEPTUNE', 'SATURN', 'URANUS', 'VENUS', '__class__', '__doc__', '__members__', '__module__']
-    >>> dir(Planet.EARTH)                   # doctest: +SKIP
```
Comment (AA-Turner) [inline Doc/howto/enum.rst:1055]: I assume removing the doctest flag is intentional?

File: Doc/library/enum.rst [hunk 4]
```diff
@@ -207,8 +207,8 @@ Data Types
 
    .. method:: EnumType.__dir__(cls)
 
-      Returns ``['__class__', '__doc__', '__members__', '__module__']`` and the
-      names of the members in *cls*::
+      Returns ``['__class__', '__contains__', '__doc__', '__getitem__', '__iter__', '__len__', '__members__', '__module__', '__name__', '__qualname__']``
+      and the names of the members in *cls*::
```
Comment (AA-Turner) [inline Doc/library/enum.rst]: I wonder if there's a better way to talk about this now that the list is longer. I'd certainly mention the names first and the special methods second.

**Review by AA-Turner — CHANGES_REQUESTED** [review]
Some initial comments:

A

File: Doc/howto/enum.rst (hunk 2 above)
Comment (ujan-r) [reply Doc/howto/enum.rst:1060]: Should I remove it? Note that there are 13 other occurrences of versionadded/versionchanged in the document

File: Misc/NEWS.d/next/Documentation/2023-08-14-20-38-39.gh-issue-107957.DU3mID.rst (hunk 1 above)
Comment (lizzydavis695) [reply Misc/NEWS.d/next/Documentation/2023-08-14-20-38-39.gh-issue-107957.DU3mID.rst]: Need help

File: Doc/howto/enum.rst (hunk 3 above)
Comment (ujan-r) [reply Doc/howto/enum.rst:1055]: No reason to leave it in now that the output is correct, right?

Comment (ujan-r) [reply Doc/howto/enum.rst:1055]: Ah, `Planet` isn't defined until later in the file. Good catch! Reverted the commit.

**Comment by ethanfurman** [comment]
These issues have been handled elsewhere -- specifically, `__dir__` was updated to *not* display the members.  Opened a new PR to acknowledge Ujan's contribution.

**Pull request closed without merging**